transport/httptransport/http_transport_perf_test.go

```diff
@@ -0,0 +1,117 @@
+package httptransport
+
+import (
+	"context"
+	"errors"
+	"io"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/filecoin-project/boost/transport/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHttpTransportMultistreamPerformance(t *testing.T) {
+	// this test runs a tcp proxy in front of the http web server that adds a latency to each tcp packet.
+	// This is done to simulate network latency and ensure that multistream transfer is reasonably faster than singlestream.
+	// rawSize and latency need to be big enough so that data transfer takes a reasonable proportion of time comparing to writing to the disk.
+	ctx := context.Background()
```

Comment:
I'd suggest we add t.Skip() at the top of this test as we will only use it occasionally.

Reply:
My idea was to have that test running for each build so that we know that nothing occasionally broke concurrent transfers. For example if the code gets changed to download chunks serially, then all existing tests would just pass while the functionality would be broken. WDYT?

Reply:
Ok seems fair, let's leave it as-is 👍